desc/compute/_profiles.py

```diff
@@ -0,0 +1,451 @@
+
+
+@register_compute_fun(
+    name="p",
+    label="p",
+    units="Pa",
+    units_long="Pascals",
+    description="Pressure",
+    dim=1,
+    params=["p_l"],
+    transforms={},
+    profiles=["pressure"],
+    function_of="r",
+    data=[],
+)
+def _p(params, transforms, profiles, data, **kwargs):
+    data["p"] = profiles["pressure"].compute(params["p_l"], dr=0)
+    return data
+
+
+@register_compute_fun(
+    name="p_r",
+    label="\\partial_{\\rho} p",
+    units="Pa",
+    units_long="Pascals",
+    description="Pressure, first radial derivative",
+    dim=1,
+    params=["p_l"],
+    transforms={},
+    profiles=["pressure"],
+    function_of="r",
+    data=[],
+)
+def _p_r(params, transforms, profiles, data, **kwargs):
+    data["p_r"] = profiles["pressure"].compute(params["p_l"], dr=1)
+    return data
+
+
+@register_compute_fun(
+    name="grad(p)",
+    label="\\nabla p",
+    units="N \\cdot m^{-3}",
+    units_long="Newtons / cubic meter",
+    description="Pressure gradient",
+    dim=3,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["p_r", "e^rho"],
+)
+def _gradp(params, transforms, profiles, data, **kwargs):
+    data["grad(p)"] = (data["p_r"] * data["e^rho"].T).T
+    return data
+
+
+@register_compute_fun(
+    name="|grad(p)|^2",
+    label="|\\nabla p|^{2}",
+    units="N^2 \\cdot m^{-6}",
+    units_long="Newtons per cubic meter squared",
+    description="Magnitude of pressure gradient squared",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["grad(p)"],
+)
+def _gradp_mag2(params, transforms, profiles, data, **kwargs):
+    data["|grad(p)|^2"] = dot(data["grad(p)"], data["grad(p)"])
+    return data
+
+
+@register_compute_fun(
+    name="|grad(p)|",
+    label="|\\nabla p|",
+    units="N \\cdot m^{-3}",
+    units_long="Newtons per cubic meter",
+    description="Magnitude of pressure gradient",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["|grad(p)|^2"],
+)
+def _gradp_mag(params, transforms, profiles, data, **kwargs):
+    data["|grad(p)|"] = jnp.sqrt(data["|grad(p)|^2"])
+    return data
+
+
+@register_compute_fun(
+    name="<|grad(p)|>_vol",
+    label="<|\\nabla p|>_{vol}",
+    units="N \\cdot m^{-3}",
+    units_long="Newtons per cubic meter",
+    description="Volume average of magnitude of pressure gradient",
+    dim=0,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    function_of="",
+    data=["|grad(p)|", "sqrt(g)", "V"],
+)
+def _gradp_mag_vol(params, transforms, profiles, data, **kwargs):
+    data["<|grad(p)|>_vol"] = (
+        jnp.sum(
+            data["|grad(p)|"] * jnp.abs(data["sqrt(g)"]) * transforms["grid"].weights
+        )
+        / data["V"]
+    )
+    return data
+
+
+@register_compute_fun(
+    name="iota",
+    label="\\iota",
+    units="~",
+    units_long="None",
+    description="Rotational transform",
```

Comment:
maybe worth putting here that this is technically iotabar (iota normalized by 2pi), this is in our docs pages but would be useful to have in this description too to be clear